pkg/registry/certificates/certificates/storage/storage.go

```diff
@@ -50,7 +52,11 @@ func NewREST(optsGetter generic.RESTOptionsGetter) (*REST, *StatusREST, *Approva
 
 		TableConvertor: printerstorage.TableConvertor{TableGenerator: printers.NewTableGenerator().With(printersinternal.AddHandlers)},
 	}
-	options := &generic.StoreOptions{RESTOptions: optsGetter}
+	options := &generic.StoreOptions{
+		RESTOptions: optsGetter,
+		AttrFunc:    csrregistry.GetAttrs,
+		TriggerFunc: map[string]storage.IndexerFunc{"spec.signerName": csrregistry.SignerNameTriggerFunc},
```

Comment:
After spending quite some time digging, I couldn't really work out if setting the TriggerFunc here is actually required... there are no doc comments on the field that I can see, nor can I see much explanation of what a TriggerFunc is used for.

I get the feeling this *may not* be required, as things like Pods support field selectors on many different fields (i.e. restartPolicy, nodeName, etc) yet only list a TriggerFunc for `spec.nodeName` (only a single element in this map is actually supported by the watch cache).

Does anyone know what the TriggerFunc is for? Does setting this here actually change anything? Should this particular change be reverted?

Reply:
it is not required. it causes the watch cache to maintain an index for this field at https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L319-L333

We typically only use it for high cardinality resources where we expect lots of filtered watchers, like nodes watching pods

Reply:
Removed